packages/react/src/createReactQueryHooks.tsx

```diff
@@ -320,12 +320,17 @@ export function createReactQueryHooks<
     ) {
       void prefetchQuery(pathAndInput as any, opts as any);
     }
-    const actualOpts = useSSRQueryOptionsIfNeeded(pathAndInput, opts);
+    const ssrOpts = useSSRQueryOptionsIfNeeded(pathAndInput, opts);
 
     return __useQuery(
       pathAndInput as any,
-      () => (client as any).query(...getClientArgs(pathAndInput, actualOpts)),
-      actualOpts,
+      ({ signal }) => {
+        const actualOpts = { ...ssrOpts, trpc: { ...ssrOpts?.trpc, signal } };
```

Comment:
I don't think we should allow passing in a signal at all for React Query. Can we not include `signal` for `TRPCReactRequestOptions`?

Reply:
I get that but right now it's possible to do this:
```ts
const abortController = new AbortController()
const signal = abortController.signal

// this aborts
trpc.post.byId.useQuery({ id: 1 }, { trpc: { signal } })

abortController.abort()
```
and I don't think this should be possible. I think we should only allow aborting through `.cancel`.

Reply:
That wouldnt work. I cc'ed you on the test that checka for this.

Reply:
Ah nice, I didn't see that!